use clean CentOS7 for monitor & use fixed AMI name in config files for loader/monitor

amoskong opened the pull request:
Let's use this public clean centos7, which is also used as base to build scylla ami. (https://github.com/scylladb/scylla/blob/master/dist/ami/build_ami.sh#L61)
I just created an instance, it's clean, no scylla stuffs.

us-east-1: ami-ae7bfdb8
us-west-1: ami-7c280d1c
us-west-2: ami-0c2aba6c
eu-west-1: ami-0d063c6b
eu-west-2: ami-c22236a6

Loader:
us-east-1: ami-050c06c3e20036edc
us-west-1: ami-0ce02e670b7ab726a
us-west-2: ami-05e91e409f39e7046
eu-west-1: ami-0006222380fb72d8d
eu-west-2: ami-0a0ab32338be1d1d0

amoskong commented:
Current monitor AMI isn't clean, it causes scylla manager testing failed.

roydahan commented:
@amoskong, can you please change also the comment from “dedicated AMI” to “clean centos”

amoskong commented:
Updated the PR:
- update comment of monitor ami to 'clean centos'

bentsi commented:
@amoskong did you test this?

amoskong commented:
I had not tested the PR in the past. I tested and updated just now, the new PR works well (Monitor/dashboard can be setup correctly)

Clean centos7 isn't enough for us, t3.small instance type requires ENA support. So I switched to another clean CentOS7 with ENA support.

Hi @yarongilor , can you help to verify the monitor AMI-id, it can satisfy requirement of scylla-manager testing?

```
    * Monitor: pass-full-centos7-2019-03-20T08-30-27Z
    us-east-1: ami-0d52990d49810f04e
    us-west-1: ami-0fdff6aad97a457ad
    us-west-2: ami-06de4d04d3603fb19
    eu-west-1: ami-092d25a6b3407805d
    eu-west-2: ami-0cd17978d178e6e3e
```

amoskong commented:
Hi @yarongilor , 

If the new monitor ami-id still doesn't work for you, we might need to process the clean AMI as scylla/dist/ami/build_ami.sh.  Or prepare a new AMI based on scylla-ami, and clean /disable scylla stuffs.

Then scylla-server can be successfully started on Monitor for testing scylla-manager.

yarongilor commented:
> Hi @yarongilor ,
> 
> If the new monitor ami-id still doesn't work for you, we might need to process the clean AMI as scylla/dist/ami/build_ami.sh. Or prepare a new AMI based on scylla-ami, and clean /disable scylla stuffs.
> 
> Then scylla-server can be successfully started on Monitor for testing scylla-manager.

a test uses ami_id_monitor: 'ami-092d25a6b3407805d' with scylla-manager passed OK.

bentsi commented on defaults/aws_config.yaml:
```diff
@@ -9,12 +9,12 @@ regions_data:
     subnet_id: 'subnet-088fddaf520e4c7a8'
     ami_id_loader: 'ami-0006222380fb72d8d' # Loader dedicated AMI
-    ami_id_monitor: 'ami-0f478f18d613e3662' # Monitor dedicated AMI
+    ami_id_monitor: 'ami-092d25a6b3407805d' # clean centos
```
Amos, where did you get those AMIs from?
they are not in official CentOS page: https://wiki.centos.org/Cloud/AWS

bentsi replied:
command to get CentOS AMIs 
`aws --region us-east-1 ec2 describe-images --owners aws-marketplace --filters Name=product-code,Values=aw0evgkw8e5c1q413zgy5pjce`

amoskong commented:
On Wed, Apr 17, 2019 at 11:33 PM Bentsi <notifications@github.com> wrote:

> *@bentsi* commented on this pull request.
> ------------------------------
>
> In defaults/aws_config.yaml
> <https://github.com/scylladb/scylla-cluster-tests/pull/904#discussion_r276303588>
> :
>
> >    eu-west-1:
>      security_group_ids: 'sg-059a7f66a947d4b5c'
>      subnet_id: 'subnet-088fddaf520e4c7a8'
>      ami_id_loader: 'ami-0006222380fb72d8d' # Loader dedicated AMI
> -    ami_id_monitor: 'ami-0f478f18d613e3662' # Monitor dedicated AMI
> +    ami_id_monitor: 'ami-092d25a6b3407805d' # clean centos
>
> Amos, where did you get those AMIs from?
> they are not in official CentOS page: https://wiki.centos.org/Cloud/AWS
>
I searched the CentOS7 (with ENA support) from aws:
-
https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#Images:visibility=public-images;search=centos7,ena;sort=desc:creationDate

amoskong commented:
On Wed, Apr 17, 2019 at 11:35 PM Bentsi <notifications@github.com> wrote:

> *@bentsi* commented on this pull request.
> ------------------------------
>
> In defaults/aws_config.yaml
> <https://github.com/scylladb/scylla-cluster-tests/pull/904#discussion_r276304891>
> :
>
> >    eu-west-1:
>      security_group_ids: 'sg-059a7f66a947d4b5c'
>      subnet_id: 'subnet-088fddaf520e4c7a8'
>      ami_id_loader: 'ami-0006222380fb72d8d' # Loader dedicated AMI
> -    ami_id_monitor: 'ami-0f478f18d613e3662' # Monitor dedicated AMI
> +    ami_id_monitor: 'ami-092d25a6b3407805d' # clean centos
>
> command to get CentOS AMIs
> aws --region us-east-1 ec2 describe-images --owners aws-marketplace
> --filters Name=product-code,Values=aw0evgkw8e5c1q413zgy5pjce
>

I found there are some AMI satisfies our requirement (clean CentOS7 & ENA
enabled) in above command outputs.
I will retest with it (ami-02eac2c0129f6376b) and update the PR.

```
        {
            "ProductCodes": [
                {
                    "ProductCodeId": "*aw0evgkw8e5c1q413zgy5pjce*",
                    "ProductCodeType": "marketplace"
                }
            ],
            "Description": "CentOS Linux 7 x86_64 HVM EBS ENA 1901_01",
            "VirtualizationType": "hvm",
            "Hypervisor": "xen",
            "ImageOwnerAlias": "aws-marketplace",
            "EnaSupport": true,
            "SriovNetSupport": "simple",
            "ImageId": "*ami-02eac2c0129f6376b*",
            "State": "available",
            "BlockDeviceMappings": [
                {
                    "DeviceName": "/dev/sda1",
                    "Ebs": {
                        "Encrypted": false,
                        "DeleteOnTermination": false,
                        "VolumeType": "gp2",
                        "VolumeSize": 8,
                        "SnapshotId": "snap-010d360e3cba720ba"
                    }
                }
            ],
            "Architecture": "x86_64",
            "ImageLocation": "aws-marketplace/CentOS Linux 7 x86_64 HVM EBS
ENA 1901_01-b7ee8a69-ee97-4a49-9e68-afaee216db2e-ami-05713873c6794f575.4",
            "RootDeviceType": "ebs",
            "OwnerId": "679593333241",
            "RootDeviceName": "/dev/sda1",
            "CreationDate": "2019-01-30T23:40:58.000Z",
            "Public": true,
            "ImageType": "machine",
            "Name": "CentOS Linux 7 x86_64 HVM EBS ENA
1901_01-b7ee8a69-ee97-4a49-9e68-afaee216db2e-ami-05713873c6794f575.4"
        },
```

>

amoskong commented:
Update:
I just updated the PR to use official AMI (with ENA support), in my test (us-east-1 & eu-west-1), the monitor setup completed successfully.

Thanks for the remind of @bentsi 

----

Hi @yarongilor , can you retest new AMI: with official AMI (with ENA support)?

    * Monitor: CentOS Linux 7 x86_64 HVM EBS ENA 1901_01
    us-east-1: ami-02eac2c0129f6376b
    us-west-1: ami-074e2d6769f445be5
    us-west-2: ami-01ed306a12b7d1c96
    eu-west-1: ami-0ff760d16d9497662
    eu-west-2: ami-0eab3a90fc693af19

amoskong commented:
I had tested the PR, the AMI-name works well, and I already switched to new ami loader
- updated cassandra_stress_exporter
- updated scylla-tools /scylla-tools-core to latest master
- removed setting of hostname

roydahan commented:
@amoskong does it include c-s-shard-aware now?

amoskong commented:
Yes.

On Thu, Apr 18, 2019 at 8:13 PM Roy Dahan <notifications@github.com> wrote:

> @amoskong <https://github.com/amoskong> does it include c-s-shard-aware
> now?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scylladb/scylla-cluster-tests/pull/904#issuecomment-484479529>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACLTTDW2FFWWYQJ6ESYPUTPRBQV7ANCNFSM4HGEHZRQ>
> .
>

bentsi commented:
@amoskong I like the idea of setting `scylla-qa-monitor-ami-latest`, `scylla-qa-loader-ami-latest`,
but how will we maintain different versions? Using this approach, we will use always the latest version of monitor and loader. For monitor it doesn't matter (since it is clean CentOS) but for loader its different.
@fruch @roydahan what do you think?

fruch commented:
I think amos idea is to rename the ami to latest each time a new version.

if we'll want to use older one we would name it, or use it's ImageId.

extending this idea we could have a branch version of the AMI if needed. for example `loader-2019.1-latest` if we want more control.

if we want a way that doesn't involves changing names of the AMIs, we can use tags to select the AMIs, and move them from version to version

amoskong commented:
I will continue maintain the AMI version by wiki:
https://github.com/scylladb/scylla-cluster-tests/wiki/SCT-AMI-maintenance-page

I have an instance (scylla-qa-loader-ami-latest (instance-id:
i-0248929691a395ca8, us-east-1)), when we have new version of AMI, I will
create a AMI in us-east-1 as backup, such as (scylla-qa-loader-ami-v4
ami-0878efb77c212f4d3)

Then copy v4 to all regions us AMI-name `scylla-qa-loader-ami-latest`


When we have v5, I will create AMI scylla-qa-loader-ami-v5, and copy the
new AMI to all region (replace old scylla-qa-loader-ami-latest)


On Thu, Apr 18, 2019 at 11:44 PM Bentsi <notifications@github.com> wrote:

> @amoskong <https://github.com/amoskong> I like the idea of setting
> scylla-qa-monitor-ami-latest, scylla-qa-loader-ami-latest,
> but how will we maintain different versions? Using this approach, we will
> use always the latest version of monitor and loader. For monitor it doesn't
> matter (since it is clean CentOS) but for loader its different.
> @fruch <https://github.com/fruch> @roydahan <https://github.com/roydahan>
> what do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scylladb/scylla-cluster-tests/pull/904#issuecomment-484567292>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACLTTCEC7NFQZG7FUHFEN3PRCJMVANCNFSM4HGEHZRQ>
> .
>

bentsi commented on sdcm/cluster_aws.py:
```diff
@@ -91,11 +90,21 @@ def __init__(self, ec2_ami_id, ec2_subnet_id, ec2_security_group_ids,
+        new_ec2_ami_id = []
+        for idx, ec2_ami in enumerate(ec2_ami_id):
+            ec2 = ec2_client.EC2Client(region_name=self.region_names[idx],
+                spot_max_price_percentage=self.params.get('spot_max_price', default=0.60))
```
`EC2Client` should have default value for spot_max_price_percentage` in the class. We should store defaults in one place.

roydahan replied:
remove this code.

roydahan commented on sdcm/ec2_client.py:
```diff
@@ -374,6 +374,16 @@ def get_instance_by_private_ip(self, private_ip):
+        if not images['Images']:
+            raise Exception('Cannot find image by name: %s' % name)
+        return images['Images'][0]['ImageId']
+
```
remove

bentsi commented:
closing this, fixed in #920
